hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -547,7 +548,11 @@ private void addContractCallResultToRecord(
 					Collections.emptyList(),
 					Collections.emptyList(),
 					EvmFnResult.EMPTY,
-					Collections.emptyMap());
+					Collections.emptyMap(),
+					precompile.shouldAddTraceabilityFieldsToRecord() ? messageFrame.getRemainingGas().toLong() : 0L,
```

Comment:
I think we should create variables for these three fields earlier in the code, at line 542.  That way we check the feature flag only once.